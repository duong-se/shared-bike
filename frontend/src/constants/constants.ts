export const tokenKey = 'accessToken'

export const commonHeaders = {
  'Accept': 'application/json',
  'Content-Type': 'application/json',
}
