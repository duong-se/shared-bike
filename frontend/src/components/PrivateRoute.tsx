import { Navigate, useLocation } from 'react-router-dom'
import { tokenKey } from '../constants/constants';

export const PrivateRoute = ({ children }: { children: JSX.Element }) => {
  const token = localStorage.getItem(tokenKey)
  let location = useLocation()
  if (!token) {
    return <Navigate to="/" replace={true} state={{ from: location }} />
  }
  return children;
}
