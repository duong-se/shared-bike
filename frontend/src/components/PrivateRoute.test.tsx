import { render, screen } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import { tokenKey } from '../constants/constants'
import { PrivateRoute } from './PrivateRoute'

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  Navigate: ({ children }: React.PropsWithChildren) => <div>{children}</div>
}));

describe('PrivateRoute', () => {
  it('should run and return children', () => {
    localStorage.setItem(tokenKey, 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MSwidXNlcm5hbWUiOiJ0ZXN0MSIsIm5hbWUiOiJUZXN0IFVzZXIiLCJwZXJtaXNzaW9ucyI6bnVsbCwiZXhwIjoxNjU3MzUyODM0fQ.jU5yp2y-3H-dxXP1hdDW-FYEYv5F9GhAVDCbafphUzI')
    render(<PrivateRoute><div data-testid="mockChildren">Test</div></PrivateRoute>, { wrapper: BrowserRouter })
    const child = screen.getByTestId('mockChildren')
    expect(child).toBeInTheDocument()
  })

  it('should run and navigate', async () => {
    localStorage.clear()
    const { container } = render(<PrivateRoute><div data-testid="mockChildren">Test</div></PrivateRoute>, { wrapper: BrowserRouter })
    expect(container).toMatchSnapshot()
  })
})
