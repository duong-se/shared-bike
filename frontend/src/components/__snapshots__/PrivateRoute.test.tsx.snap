// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute should run and navigate 1`] = `
<div>
  <div />
</div>
`;
